rfd/0057-automatic-aws-server-discovery.md

```diff
@@ -235,6 +235,153 @@ discovery_service:
     teleport.dev/origin: "cloud"
 ```
 
+
+### Agentless installation
+
+In addition to supporting automatic Teleport agent installation, an agentless option
+will also be supported. This mode will update the OpenSSH CA to use the Teleport CA
+without installing the full Teleport Agent.
+
+The teleport ca and host keys will be generated and then uploaded to
```

Comment:
@klizhentas We're not going to use SSM to send private keys. The idea is to use Secrets Manager: the discovery service will use Secrets Manager to store nodes' credentials and nodes will fetch them from there.

For AWS joining we'd need a Teleport process on the node but this is "agentless" mode, or am I misunderstanding what you're proposing? Are you thinking of installing a `teleport` binary on the nodes just to have it go through the AWS join method and update OpenSSH config, but not actually install it as a service?

Reply:
> Are you thinking of installing a teleport binary on the nodes just to have it go through the AWS join method and update OpenSSH config, but not actually install it as a service?

Right, so we can re-use exactly the same flow we had already built

Reply:
@klizhentas In that case what do you think about introducing a command e.g. `teleport join` that would:

- Fetch node credentials using regular Teleport join method (IAM in this case).
- Update OpenSSH config with host certs and user CA.
- Restart sshd.

E.g.

```
teleport join \
  --proxy=proxy.example.com \
  --token=iam-auth-token \
  [--openssh-config-path=/path/to/config] \
  [--restart-sshd=true]
```

This way we won't have to build all this logic in a bash script, it will be just a Go tool built into Teleport (similar to what we did with `teleport install systemd`). And we'll probably also be able to simplify our OpenSSH guides a bit to use this command.